File: Java/Dagger/atm/src/main/java/ro/razvan/java/dagger/atm/SingleArgCommand.java
package ro.razvan.java.dagger.atm;

import java.util.List;

public interface SingleArgCommand extends Command {

    @Override
    default Result handleInput(List<String> input) {
        if (input.size() != 1) return Result.invalid();

        return handleArg(input.get(0));
    }

    Result handleArg(String arg);

}

File: Kotlin/Dagger/atm/src/main/kotlin/ro/razvan/kotlin/dagger/atm/SingleArgCommand.kt
package ro.razvan.kotlin.dagger.atm

import ro.razvan.kotlin.dagger.atm.Command.Result

interface SingleArgCommand : Command {

    override fun handleInput(input: List<String>): Result {
        if (input.size != 1) return Result.invalid()

        return handleArg(input[0])
    }

    fun handleArg(arg: String): Result

}